src/services/goToDefinition.ts

```diff
@@ -2,8 +2,10 @@
 namespace ts.GoToDefinition {
     export function getDefinitionAtPosition(program: Program, sourceFile: SourceFile, position: number): readonly DefinitionInfo[] | undefined {
         const resolvedRef = getReferenceAtPosition(sourceFile, position, program);
-        if (resolvedRef) {
-            return [getDefinitionInfoForFileReference(resolvedRef.reference.fileName, resolvedRef.file.fileName)];
+        const fileReferenceDefinition = resolvedRef && [getDefinitionInfoForFileReference(resolvedRef.reference.fileName, resolvedRef.fileName, !!resolvedRef.file)] || emptyArray;

```

Comment:
Having source file present is not good enough indicator of "guess" as `resolvePath(getDirectoryPath(sourceFile.fileName), node.text);` this is just a guess. so wether file exists or not it should be marked as unverified?